File: examples/micronaut-app/src/test/java/ht/eyfout/example/http/ExampleWhenScope.java
package ht.eyfout.example.http;

import ht.eyfout.junit.jupiter.api.GivenState;
import ht.eyfout.junit.jupiter.api.http.HttpAPIWhenScope;

public class ExampleWhenScope extends HttpAPIWhenScope {

    public ExampleWhenScope(GivenState givenState) {
        super(givenState);
    }

    public String getAuthorization() {
        return ((ExampleGivenState) givenState).authorization;
    }

    public String getManufacturerID() {
        return ((ExampleGivenState) givenState).manufacturerID;
    }

    public String getVehicleID() {
        return ((ExampleGivenState) givenState).vehicleID;
    }

}

File: examples/micronaut-app/src/test/kotlin/ht/eyfout/example/http/ExampleWhenScope.kt
package ht.eyfout.example.http

import ht.eyfout.junit.jupiter.api.GivenState
import ht.eyfout.junit.jupiter.api.http.HttpAPIWhenScope

class ExampleWhenScope(givenState: GivenState): HttpAPIWhenScope(givenState) {

    val authorization: String
        get() = (givenState as ExampleGivenState).authorization
    val manufacturerID: String
        get() = (givenState as ExampleGivenState).manufacturerID
    val vehicleID: String
        get() = (givenState as ExampleGivenState).vehicleID

}